parsers/IL.py

```diff
@@ -55,14 +55,46 @@ def flatten_list(_2d_list) -> list:
     return flatten_list(cleaned_list)
 
 
+def fetch_price_date():
+    """Fetch updated price date."""
+    response = get(IEC_PRICE)
+    soup = BeautifulSoup(response.content, "lxml")
+
+    date_str = soup.find("span", lang="HE").text
+    date_str = date_str.split(sep=" - ")
```

Comment:
It seems like this method does another call to the URL `IEC_PRICE`.
Why not have it pass the `soup` object from L78?
Then you only do a single URL call.
```suggestion
def parse_price_date_from_soup(soup):
    """Fetch updated price date."""

    date_str = soup.find("span", lang="HE").text
    date_str = date_str.split(sep=" - ")
```

Reply:
Did something a bit different but I hope that is fine. did not figure out how to pass that argument and still call the function once..